File: CS/MeffedNavigationService.Infrastructure/Contract/ExportViewAttribute.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.ComponentModel;
using System.ComponentModel.Composition;

namespace MeffedNavigationService.Infrastructure.Contract {
    [MetadataAttribute]
    [AttributeUsage(AttributeTargets.Class, AllowMultiple = false)]
    public class ExportViewAttribute : ExportAttribute {
        /// <summary>
        /// Initializes a new instance of the ExportViewAttribute class.
        /// </summary>
        public ExportViewAttribute()
            : base(typeof(IModule)) {
        }
        public MeffedNavigationService.Infrastructure.Enums.ModuleTypes ModuleType { get; set; }
    }
}

File: VB/MeffedNavigationService.Infrastructure/Contract/ExportViewAttribute.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.ComponentModel
Imports System.ComponentModel.Composition

Namespace MeffedNavigationService.Infrastructure.Contract
	<MetadataAttribute, AttributeUsage(AttributeTargets.Class, AllowMultiple := False)> _
	Public Class ExportViewAttribute
		Inherits ExportAttribute
		''' <summary>
		''' Initializes a new instance of the ExportViewAttribute class.
		''' </summary>
		Public Sub New()
			MyBase.New(GetType(IModule))
		End Sub
		Private privateModuleType As MeffedNavigationService.Infrastructure.Enums.ModuleTypes
		Public Property ModuleType() As MeffedNavigationService.Infrastructure.Enums.ModuleTypes
			Get
				Return privateModuleType
			End Get
			Set(ByVal value As MeffedNavigationService.Infrastructure.Enums.ModuleTypes)
				privateModuleType = value
			End Set
		End Property
	End Class
End Namespace